microprofile/websocket/src/test/java/io/helidon/microprofile/tyrus/WebSocketBaseTest.java

```diff
@@ -26,71 +30,33 @@
 import javax.websocket.OnMessage;
 import javax.websocket.OnOpen;
 import javax.websocket.Session;
-import javax.websocket.server.ServerApplicationConfig;
 import javax.websocket.server.ServerEndpoint;
-import javax.websocket.server.ServerEndpointConfig;
-
-import java.io.IOException;
-import java.net.URI;
-import java.util.Collections;
-import java.util.Set;
-import java.util.logging.Logger;
 
-import io.helidon.microprofile.server.RoutingPath;
-import io.helidon.microprofile.server.Server;
 import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
 /**
- * Class WebSocketAppTest.
+ * Class WebSocketBaseTest.
  */
-public class WebSocketAppTest {
-
-    private static Server server;
+public abstract class WebSocketBaseTest {
 
-    @BeforeAll
-    static void initClass() {
-        Server.Builder builder = Server.builder();
-        server = builder.build();
-        server.start();
-    }
+    static final int DEFAULT_PORT = 7001;
+    static SeContainer container;
 
     @AfterAll
     static void destroyClass() {
-        server.stop();
+        container.close();
     }
 
-    @Test
-    public void testEchoAnnot() throws Exception {
-        URI echoUri = URI.create("ws://localhost:" + server.port() + "/web/echoAnnot");
-        EchoClient echoClient = new EchoClient(echoUri);
-        echoClient.echo("hi", "how are you?");
-    }
+    public abstract String context();
 
     @Test
-    public void testEchoProg() throws Exception {
-        URI echoUri = URI.create("ws://localhost:" + server.port() + "/web/echoProg");
+    public void testEchoAnnot() throws Exception {
+        URI echoUri = URI.create("ws://localhost:" + DEFAULT_PORT + context() + "/echoAnnot");
```

Comment:
I know there have been discussions about avoiding fixed ports because of conflicts on shared systems or parallel tests. Is this opening us up to that problem?

Reply:
Possibly. I had to switch the startup code to use CDI so I can register beans directly and I don't have access to the server port anymore. The old test was always starting on the default port anyway. So it shouldn't be worse than before, but it does not mean is great.